Lib/unittest/mock.py

```diff
@@ -2482,6 +2485,52 @@ def __set__(self, obj, val):
         self(val)
 
 
+class WaitableMock(MagicMock):
+    """
+    A mock that can be used to wait until it was called.
+
+    `event_class` - Class to be used to create event object.
+    Defaults to Threading.Event and can take values like multiprocessing.Event.
+    """
+
+    def __init__(self, *args, event_class=threading.Event, **kwargs):
+        _safe_super(WaitableMock, self).__init__(*args, **kwargs)
+        self._event_class = event_class
+        self._event = event_class()
+        self._expected_calls = defaultdict(event_class)
+
+    def _mock_call(self, *args, **kwargs):
+        ret_value = _safe_super(WaitableMock, self)._mock_call(*args, **kwargs)
+
+        for call in self._mock_mock_calls:
+            event = self._expected_calls[call.args]
+            event.set()
+
+        self._event.set()
+
+        return ret_value
+
+    def wait_until_called(self, timeout=1.0):
+        """Wait until the mock object is called.
+
+        `timeout` - time to wait for in seconds. Defaults to 1.
+        """
+        return self._event.wait(timeout=timeout)
+
+    def wait_until_called_with(self, *args, timeout=1.0):
+        """Wait until the mock object is called with given args.
+
+        `timeout` - time to wait for in seconds. Defaults to 1.
+        """
+        if args not in self._expected_calls:
```

Comment:
You dont need this anymore thanks to the default dict, these 3 lines are identical to just `event = self._expected_calls[args]`.